docs/develop/python/message-passing.mdx

```diff
@@ -27,25 +27,27 @@ tags:
   - dynamic-handlers
 ---
 
-A Workflow can be thought of as a stateful web service that can receive messages: Queries, Signals, and Updates.
-Clients can use these to read data from a Workflow and control its execution.
-See [Workflow message passing](/encyclopedia/workflow-message-passing) for a non–Python-specific
-introduction.
+A Workflow can act like a stateful web service that receives messages: Queries, Signals, and Updates.
+The Workflow implementation defines these endpoints via handler methods that can react to incoming messages and return values.
+Temporal Clients use messages to read Workflow state and control its execution.
+See [Workflow message passing](/encyclopedia/workflow-message-passing) for a general overview of this topic.
+This page introduces these features for the Temporal Python SDK.
 
-## Writing message handlers
+## Write message handlers {#writing-message-handlers}
 
-The code samples below form part of a complete working sample, viewable at [message_passing/introduction](https://github.com/temporalio/samples-python/tree/message-passing/message_passing/introduction).
+:::info
+The code that follows is part of a working message passing [sample](https://github.com/temporalio/samples-python/tree/message-passing/message_passing/introduction).
+:::
 
-General guidelines for writing message handlers:
+Follow these guidelines when writing your message handlers:
```

Comment:
The following bullet arose out of a thread with @axfelix who asked to make it clearer where handlers are defined. The bullet got lost in branch mechanics but do you like it?

```suggestion
Follow these guidelines when writing your message handlers:

- Message handlers are defined as methods on the Workflow class, using one of the three decorators: [`@workflow.query`](https://python.temporal.io/temporalio.workflow.html#query), [`@workflow.signal`](https://python.temporal.io/temporalio.workflow.html#signal), and [`@workflow.update`](https://python.temporal.io/temporalio.workflow.html#update).
```